### DISCUSSION TITLE
Put this in Issues, but it may be better off here

### DISCUSSION THREAD

**Discussion by JoshKerman** [hgrecco/pint#1852] [Q&A] [answered]
When importing pint, util.py throws "TypeError: 'ABCMeta' object is not subscriptable" as shown.
Using an up-to-date Python 3.11 interpreter and pint 0.22.

Are there any changes I can make to the pint code to make it work?

![image_2023-09-21_153619226](https://github.com/hgrecco/pint/assets/145699464/781bfbc0-8d29-4939-b235-741cbffc2bba)

**Comment by hgrecco** [comment]
There is a potential solution here: https://stackoverflow.com/questions/59955751/abcmeta-object-is-not-subscriptable-when-trying-to-annotate-a-hash-variable

**Reply by JoshKerman** [reply]
I was looking at that, too, but the error seems to be caused by collection.abc.Iterable
Might replacing collection.abc.Iterable with typing.Sequence work?

**Reply by hgrecco** [reply, marked as answer]
By typing.Iterable according to that post. But I still do not understand why it works with our tests. I guess Spyder is checking types in a way our code is not.

**Reply by JoshKerman** [reply]
It might just be my machine, though. I need to use pint for a class, and most of my classmates installed using Anaconda, and it works fine. I just _had_ to install spyder and pint bare-bones.

**Reply by JoshKerman** [reply]
Changing Iterable from collections to typing seems to have resolved the issue in question, but now, a new TypeError is happening in the line right below that. I will begin looking into the new error.

![image_2023-09-21_160043234](https://github.com/hgrecco/pint/assets/145699464/6b891f07-1a20-4cef-a9d6-3b498dde6a04)

**Reply by JoshKerman** [reply]
Yeah. . . I still have no idea why Matrix = list[list[scalar]] is throwing a TypeError.

**Comment by hgrecco** [comment]
What I do not understand is why this is not failing in our tests.
